### PR TITLE
[Diagnostics] Add environment variable for controlling top-level printing and fix issue with pretty printing/parsing roundtrip.

### PR DESCRIPTION
**Pull request by jroesch**
Fix some parsing issues from NMS and add top-level environment variables. cc @electriclilies @mbrookhart @jwfromm @tkonolige

### CODE REVIEW

File: python/tvm/__init__.py [hunk 1]
```diff
@@ -67,16 +67,33 @@
...
 
+def _should_print_backtrace():
+    in_pytest = "PYTEST_CURRENT_TEST" in os.environ
+    tvm_backtrace = os.environ.get("TVM_BACKTRACE", "0")
+
+    try:
+        tvm_backtrace = bool(int(tvm_backtrace))
+    except ValueError:
+        raise ValueError(
+            "invalid value for TVM_BACKTRACE `{tvm_backtrace}`, please set to 0 or 1.")
```
Comment (tkonolige) [inline python/tvm/__init__.py:78]:
```suggestion
            f"invalid value for TVM_BACKTRACE `{tvm_backtrace}`, please set to 0 or 1.")
```

File: python/tvm/__init__.py [hunk 2]
```diff
@@ -67,16 +67,33 @@
...
+
+    return in_pytest or tvm_backtrace
 
 def tvm_wrap_excepthook(exception_hook):
     """Wrap given excepthook with TVM additional work."""
 
     def wrapper(exctype, value, trbk):
         """Clean subprocesses when TVM is interrupted."""
         in_pytest = "PYTEST_CURRENT_TEST" in os.environ
+        tvm_backtrace = os.environ.get("TVM_BACKTRACE", "0")
```
Comment (tkonolige) [inline python/tvm/__init__.py:88]: I think you want to delete these lines

File: src/parser/parser.cc [hunk 3]
```diff
@@ -605,30 +605,42 @@ class Parser {
     return ast;
   }
 
+  struct MetaRef {
```
Comment (tkonolige) [inline src/parser/parser.cc:608]: Are changes in this file relevant?

Comment (jroesch) [reply src/parser/parser.cc:608]: Yes they fix a bug in the parsing which prevents this from working on all source files. I was going to turn on annotate spans for all models.

**Review by mbrookhart — APPROVED** [review]
Thanks, Jared, this fixes a number of issues I've seen in complicated models. 

@tkonolige Do you want to take another look?

**Comment by mbrookhart** [comment]
Thanks @jroesch @tkonolige !

**Pull request merged**